### PR TITLE
Create generic appointment summary component

### PR DESCRIPTION
**Pull request by andrewburrowsmoj**
## What does this pull request do?

Create a dedicated class for providing Appointment Summary details.

## What is the intent behind these changes?

Have a single place for defining the appointment summary.

This component will be used in the following classes:

AttendanceFeedbackPresenter
CheckFeedbackAnswersPresenter
ScheduleAppointmentView (when rescheduled - not yet implemented)
SubmittedFeedbackPresenter
SupplierAssessmentAppointmentPresenter

I wanted to do this refactoring as part of my work on rescheduling an initial assessment appointment. As part of that work I need to provide an appointment summary. I noticed that this is duplicated in multiple places and has some inconsistencies.

After higher priority work has finished I will work on changing the above classes to use this component.

The appointment summary looks like this:
image

### CODE REVIEW

File: server/routes/appointments/appointmentSummaryComponent.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { SummaryListItem } from '../../utils/summaryList'
+import { ActionPlanAppointment } from '../../models/actionPlan'
+import Appointment from '../../models/appointment'
+import AppointmentDecorator from '../../decorators/appointmentDecorator'
+import PresenterUtils from '../../utils/presenterUtils'
+import AuthUserDetails from '../../models/hmppsAuth/authUserDetails'
+
+export default class AppointmentSummaryComponent {
```
Comment (Gweaton) [inline server/routes/appointments/appointmentSummaryComponent.ts:8]: I think we can drop the `Component` from the name here as we don't tend to use `component` much in our codebase - I initially thought `AppointmentSummaryPresenter` might be better, but equally `AppointmentSummary` could work on its own? What do you think?

File: server/routes/appointments/appointmentSummaryComponent.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  constructor(
+    private readonly appointment: Appointment | ActionPlanAppointment,
+    private readonly assignedCaseworker: AuthUserDetails | null = null
+  ) {}
+
+  private readonly appointmentDecorator = new AppointmentDecorator(this.appointment)
+
+  get appointmentSummary(): SummaryListItem[] {
```
Comment (Gweaton) [inline server/routes/appointments/appointmentSummaryComponent.ts:16]: maybe we can call this `.summaryList` if we rename the class as suggested above.

**Review by Gweaton — COMMENTED** [review]
Looks good - just one thing re: naming.

File: server/routes/appointments/appointmentSummaryComponent.ts (hunk 2 above)
Comment (andrewburrowsmoj) [reply server/routes/appointments/appointmentSummaryComponent.ts]: I've renamed to appointmentSummaryList

File: server/routes/appointments/appointmentSummaryComponent.ts (hunk 1 above)
Comment (andrewburrowsmoj) [reply server/routes/appointments/appointmentSummaryComponent.ts]: (we do have AddressFromComponent -  I don't think it matters much, but having a consistent way to differentiate between something that is 1-2-1 linked to a page and just a general component for use would be nice)

I've renamed to AppointmentSummary

**Review by Gweaton — APPROVED** [review]
Thanks for the changes!

**Pull request merged**